src/awkward/_connect/numexpr.py

```diff
@@ -15,10 +16,12 @@
 from awkward._namedaxis import NAMED_AXIS_KEY, NamedAxesWithDims, _unify_named_axis
 
 _has_checked_version = False
+_numexpr_version_ge_2_11_0 = False
 
 
 def _import_numexpr():
     global _has_checked_version
+    global _numexpr_version_ge_2_11_0
```

Comment:
Fine for now, since it's following the same pattern as something else, but it's not the right way to do this. The correct way is to make this a function and add `@functools.cache` to it. That will not require `global` and will also be threadsafe.